pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java

```diff
@@ -93,6 +90,14 @@ public String toExplainString() {
 
   @Override
   protected TransferableBlock getNextBlock() {
+    if (!_statsCollected) {
```

Comment:
This change was made cause operatorIndex is null when this method is called from the constructor.

Reply:
could you explain? how can this result in operatorIndex being null?

Reply:
operatorIndex is set after the LeafOperator is created. 
However, we were recording the stats in the map during the constructor itself.